# Modelling Potential RNA:RNA Interactions Between SARS-CoV-2 and Human Transcriptome

## *Pitch*
Need a pitch

## *Description*
Given the short timescale between when SARS-Cov-2 was first discovered and when it was designated a pandemic, there is next-to-no high-throughput data for the virus. This precludes in-depth integrative, systems-level analyses on it and as such our understanding of the large-scale effects it has on cellular behaviour are limited. However, there is such data for other coronaviruses, namely the 2003 outbreak strain (which SARS-CoV-2 is believed to be closely related to) and Middle Eastern Respiratory Syndrome (MERS) virus, alongside other less well-documented coronaviruses. Using these data, differential expression analyses and various gene regulatory network construction methods, it may be possible to identify "consensus" modules/pathways that are dysregulated by all coronaviruses, or at least in coronaviruses that are highly related (SARS coronaviruses). Modules/pathways identified this way could then be compared to the current SARS-CoV-2 literature to identify any evidence that these gene sets are dys-regulated during SARS-CoV-2 infection, with the aim of providing better understanding of the life cycle of the virus and how the virus causes COVID-19.

## *Objectives*
* 

## *The Team*
** Amber Paulson (Team Lead)**
Description

** Name (Pipline Construction Team Member)**


**Name (Visulization/Anlysis Team Member)**
Description

**Name (Integrated Management Team Member)**
Description

## Useful Resources 

1. Non-canonical SARS CoV-2 sgRNAs from UCSC browser deposit by Kim et al., 2020 
2. Human 3’UTR data from  UTRdb database 
3. Human 5’UTR data from the EnhancerAtlas database


### Bioinformatics Approach and Tool Used

## Visualization Approach Tools Used

## References
